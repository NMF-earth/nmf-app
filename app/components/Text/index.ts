import H1 from "./H1";
import H2 from "./H2";
import H3 from "./H3";
import Primary from "./Primary";
import Secondary from "./Secondary";
import Tertiary from "./Tertiary";
import Link from "./Link";

export default { H1, H2, H3, Primary, Secondary, Tertiary, Link };
