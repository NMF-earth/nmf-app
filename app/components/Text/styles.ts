import { StyleSheet } from "react-native";
import colors from "../../style/colors";
import { fontWeight } from "../../style/font";

export default StyleSheet.create({
  darkGray: {
    color: colors.darkLink70
  },
  lightGray: {
    color: colors.swordGray
  },
  green: {
    color: colors.linkGreen
  },
  white: {
    color: colors.white
  },
  bold: {
    fontFamily: fontWeight.Bold
  },
  center: {
    textAlign: "center"
  }
});
