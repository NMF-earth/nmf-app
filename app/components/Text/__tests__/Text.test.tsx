import React from "react";
import { create } from "react-test-renderer";
import * as WebBrowser from "expo-web-browser";

import Text from "../";

jest.unmock("../");

/* H1 Tests */

it("renders correctly H1", () => {
  const tree = create(<Text.H1>test</Text.H1>).toJSON();
  expect(tree).toMatchSnapshot();
});

it("renders correctly H1 darkGray", () => {
  const tree = create(<Text.H1 darkGray>test</Text.H1>).toJSON();
  expect(tree).toMatchSnapshot();
});

it("renders correctly H1 lightGray", () => {
  const tree = create(<Text.H1 lightGray>test</Text.H1>).toJSON();
  expect(tree).toMatchSnapshot();
});

it("renders correctly H1 green", () => {
  const tree = create(<Text.H1 green>test</Text.H1>).toJSON();
  expect(tree).toMatchSnapshot();
});

/* H2 Tests */

it("renders correctly H2", () => {
  const tree = create(<Text.H2>test</Text.H2>).toJSON();
  expect(tree).toMatchSnapshot();
});

it("renders correctly H2 darkGray", () => {
  const tree = create(<Text.H2 darkGray>test</Text.H2>).toJSON();
  expect(tree).toMatchSnapshot();
});

it("renders correctly H2 blue50", () => {
  const tree = create(<Text.H2 blue50>test</Text.H2>).toJSON();
  expect(tree).toMatchSnapshot();
});

it("renders correctly H2 lightGray", () => {
  const tree = create(<Text.H2 lightGray>test</Text.H2>).toJSON();
  expect(tree).toMatchSnapshot();
});

it("renders correctly H2 green", () => {
  const tree = create(<Text.H2 green>test</Text.H2>).toJSON();
  expect(tree).toMatchSnapshot();
});

/* H3 Tests */

it("renders correctly H3", () => {
  const tree = create(<Text.H3>test</Text.H3>).toJSON();
  expect(tree).toMatchSnapshot();
});

it("renders correctly H3 darkGray", () => {
  const tree = create(<Text.H3 darkGray>test</Text.H3>).toJSON();
  expect(tree).toMatchSnapshot();
});

it("renders correctly H3 lightGray", () => {
  const tree = create(<Text.H3 lightGray>test</Text.H3>).toJSON();
  expect(tree).toMatchSnapshot();
});

it("renders correctly H3 green", () => {
  const tree = create(<Text.H3 green>test</Text.H3>).toJSON();
  expect(tree).toMatchSnapshot();
});

it("renders correctly H3 center", () => {
  const tree = create(<Text.H3 center>test</Text.H3>).toJSON();
  expect(tree).toMatchSnapshot();
});

/* Primary Tests */

it("renders correctly Primary", () => {
  const tree = create(<Text.Primary>test</Text.Primary>).toJSON();
  expect(tree).toMatchSnapshot();
});

it("renders correctly Primary darkGray", () => {
  const tree = create(<Text.Primary darkGray>test</Text.Primary>).toJSON();
  expect(tree).toMatchSnapshot();
});

it("renders correctly Primary lightGray", () => {
  const tree = create(<Text.Primary lightGray>test</Text.Primary>).toJSON();
  expect(tree).toMatchSnapshot();
});

it("renders correctly Primary green", () => {
  const tree = create(<Text.Primary green>test</Text.Primary>).toJSON();
  expect(tree).toMatchSnapshot();
});

it("renders correctly Primary white", () => {
  const tree = create(<Text.Primary white>test</Text.Primary>).toJSON();
  expect(tree).toMatchSnapshot();
});

it("renders correctly Primary bold", () => {
  const tree = create(<Text.Primary bold>test</Text.Primary>).toJSON();
  expect(tree).toMatchSnapshot();
});

it("renders correctly Primary center", () => {
  const tree = create(<Text.Primary bold>test</Text.Primary>).toJSON();
  expect(tree).toMatchSnapshot();
});

it("renders correctly Primary red", () => {
  const tree = create(<Text.Primary red>test</Text.Primary>).toJSON();
  expect(tree).toMatchSnapshot();
});

/* Primary Tests */

it("renders correctly Secondary", () => {
  const tree = create(<Text.Secondary>test</Text.Secondary>).toJSON();
  expect(tree).toMatchSnapshot();
});

it("renders correctly Secondary darkGray", () => {
  const tree = create(<Text.Secondary darkGray>test</Text.Secondary>).toJSON();
  expect(tree).toMatchSnapshot();
});

it("renders correctly Secondary lightGray", () => {
  const tree = create(<Text.Secondary lightGray>test</Text.Secondary>).toJSON();
  expect(tree).toMatchSnapshot();
});

it("renders correctly Secondary green", () => {
  const tree = create(<Text.Secondary green>test</Text.Secondary>).toJSON();
  expect(tree).toMatchSnapshot();
});

it("renders correctly Secondary white", () => {
  const tree = create(<Text.Secondary white>test</Text.Secondary>).toJSON();
  expect(tree).toMatchSnapshot();
});

it("renders correctly Secondary bold", () => {
  const tree = create(<Text.Secondary bold>test</Text.Secondary>).toJSON();
  expect(tree).toMatchSnapshot();
});

it("renders correctly Secondary center", () => {
  const tree = create(<Text.Secondary center>test</Text.Secondary>).toJSON();
  expect(tree).toMatchSnapshot();
});

it("renders correctly Secondary lightWeightText", () => {
  const tree = create(
    <Text.Secondary lightWeightText>test</Text.Secondary>
  ).toJSON();
  expect(tree).toMatchSnapshot();
});

it("renders correctly Secondary red", () => {
  const tree = create(<Text.Secondary red>test</Text.Secondary>).toJSON();
  expect(tree).toMatchSnapshot();
});

/* Tertiary Tests */

it("renders correctly Tertiary", () => {
  const tree = create(<Text.Tertiary>test</Text.Tertiary>).toJSON();
  expect(tree).toMatchSnapshot();
});

it("renders correctly Tertiary darkGray", () => {
  const tree = create(<Text.Tertiary darkGray>test</Text.Tertiary>).toJSON();
  expect(tree).toMatchSnapshot();
});

it("renders correctly Tertiary lightGray", () => {
  const tree = create(<Text.Tertiary lightGray>test</Text.Tertiary>).toJSON();
  expect(tree).toMatchSnapshot();
});

it("renders correctly Tertiary green", () => {
  const tree = create(<Text.Tertiary green>test</Text.Tertiary>).toJSON();
  expect(tree).toMatchSnapshot();
});

it("renders correctly Tertiary white", () => {
  const tree = create(<Text.Tertiary white>test</Text.Tertiary>).toJSON();
  expect(tree).toMatchSnapshot();
});

it("renders correctly Tertiary bold", () => {
  const tree = create(<Text.Tertiary bold>test</Text.Tertiary>).toJSON();
  expect(tree).toMatchSnapshot();
});

it("renders correctly Tertiary center", () => {
  const tree = create(<Text.Tertiary center>test</Text.Tertiary>).toJSON();
  expect(tree).toMatchSnapshot();
});

it("renders correctly Tertiary lightWeightText", () => {
  const tree = create(
    <Text.Tertiary lightWeightText>test</Text.Tertiary>
  ).toJSON();
  expect(tree).toMatchSnapshot();
});

it("renders correctly Tertiary red", () => {
  const tree = create(<Text.Tertiary red>test</Text.Tertiary>).toJSON();
  expect(tree).toMatchSnapshot();
});

/* Link Tests */

it("renders correctly Link", () => {
  const tree = create(
    <Text.Link url="http://nmf.earth">test</Text.Link>
  ).toJSON();
  expect(tree).toMatchSnapshot();
});

it("open links", () => {
  const spy = jest.spyOn(WebBrowser, "openBrowserAsync");
  const root = create(<Text.Link url="http://nmf.earth">test</Text.Link>).root;
  // eslint-disable-next-line @typescript-eslint/ban-ts-ignore
  // @ts-ignore
  const btn = root.findByType("TouchableOpacity");
  btn.props.onPress();
  expect(spy).toHaveBeenCalled();
});
