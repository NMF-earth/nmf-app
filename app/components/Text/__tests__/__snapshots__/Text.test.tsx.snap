// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly H1 1`] = `
<Text
  style={
    Array [
      Object {
        "color": "#111714",
        "fontFamily": "Inter-ExtraBold",
        "fontSize": 30,
      },
      undefined,
    ]
  }
>
  test
</Text>
`;

exports[`renders correctly H1 darkGray 1`] = `
<Text
  darkGray={true}
  style={
    Array [
      Object {
        "color": "#111714",
        "fontFamily": "Inter-ExtraBold",
        "fontSize": 30,
      },
      undefined,
      Object {
        "color": "#585D5B",
      },
    ]
  }
>
  test
</Text>
`;

exports[`renders correctly H1 lightGray 1`] = `
<Text
  lightGray={true}
  style={
    Array [
      Object {
        "color": "#111714",
        "fontFamily": "Inter-ExtraBold",
        "fontSize": 30,
      },
      undefined,
      Object {
        "color": "#9CA6A0",
      },
    ]
  }
>
  test
</Text>
`;

exports[`renders correctly H2 1`] = `
<Text
  style={
    Array [
      Object {
        "color": "#111714",
        "fontFamily": "Inter-Bold",
        "fontSize": 26,
      },
      undefined,
    ]
  }
>
  test
</Text>
`;

exports[`renders correctly H2 darkGray 1`] = `
<Text
  darkGray={true}
  style={
    Array [
      Object {
        "color": "#111714",
        "fontFamily": "Inter-Bold",
        "fontSize": 26,
      },
      undefined,
      Object {
        "color": "#585D5B",
      },
    ]
  }
>
  test
</Text>
`;

exports[`renders correctly H2 lightGray 1`] = `
<Text
  lightGray={true}
  style={
    Array [
      Object {
        "color": "#111714",
        "fontFamily": "Inter-Bold",
        "fontSize": 26,
      },
      undefined,
      Object {
        "color": "#9CA6A0",
      },
    ]
  }
>
  test
</Text>
`;

exports[`renders correctly H3 1`] = `
<Text
  style={
    Array [
      Object {
        "color": "#111714",
        "fontFamily": "Inter-Bold",
        "fontSize": 22,
      },
      undefined,
    ]
  }
>
  test
</Text>
`;

exports[`renders correctly H3 darkGray 1`] = `
<Text
  darkGray={true}
  style={
    Array [
      Object {
        "color": "#111714",
        "fontFamily": "Inter-Bold",
        "fontSize": 22,
      },
      undefined,
      Object {
        "color": "#585D5B",
      },
    ]
  }
>
  test
</Text>
`;

exports[`renders correctly H3 lightGray 1`] = `
<Text
  lightGray={true}
  style={
    Array [
      Object {
        "color": "#111714",
        "fontFamily": "Inter-Bold",
        "fontSize": 22,
      },
      undefined,
      Object {
        "color": "#9CA6A0",
      },
    ]
  }
>
  test
</Text>
`;

exports[`renders correctly Primary 1`] = `
<Text
  style={
    Array [
      Object {
        "color": "#111714",
        "fontFamily": "Inter-Medium",
        "fontSize": 18,
      },
      undefined,
    ]
  }
>
  test
</Text>
`;

exports[`renders correctly Primary bold 1`] = `
<Text
  bold={true}
  style={
    Array [
      Object {
        "color": "#111714",
        "fontFamily": "Inter-Medium",
        "fontSize": 18,
      },
      undefined,
      Object {
        "fontFamily": "Inter-Bold",
      },
    ]
  }
>
  test
</Text>
`;

exports[`renders correctly Primary center 1`] = `
<Text
  bold={true}
  style={
    Array [
      Object {
        "color": "#111714",
        "fontFamily": "Inter-Medium",
        "fontSize": 18,
      },
      undefined,
      Object {
        "fontFamily": "Inter-Bold",
      },
    ]
  }
>
  test
</Text>
`;

exports[`renders correctly Primary darkGray 1`] = `
<Text
  darkGray={true}
  style={
    Array [
      Object {
        "color": "#111714",
        "fontFamily": "Inter-Medium",
        "fontSize": 18,
      },
      undefined,
      Object {
        "color": "#585D5B",
      },
    ]
  }
>
  test
</Text>
`;

exports[`renders correctly Primary green 1`] = `
<Text
  green={true}
  style={
    Array [
      Object {
        "color": "#111714",
        "fontFamily": "Inter-Medium",
        "fontSize": 18,
      },
      undefined,
      Object {
        "color": "#4BB377",
      },
    ]
  }
>
  test
</Text>
`;

exports[`renders correctly Primary lightGray 1`] = `
<Text
  lightGray={true}
  style={
    Array [
      Object {
        "color": "#111714",
        "fontFamily": "Inter-Medium",
        "fontSize": 18,
      },
      undefined,
      Object {
        "color": "#9CA6A0",
      },
    ]
  }
>
  test
</Text>
`;

exports[`renders correctly Primary white 1`] = `
<Text
  style={
    Array [
      Object {
        "color": "#111714",
        "fontFamily": "Inter-Medium",
        "fontSize": 18,
      },
      undefined,
      Object {
        "color": "#FFFFFF",
      },
    ]
  }
  white={true}
>
  test
</Text>
`;

exports[`renders correctly Secondary 1`] = `
<Text
  style={
    Array [
      Object {
        "color": "#111714",
        "fontFamily": "Inter-Medium",
        "fontSize": 16,
      },
      undefined,
    ]
  }
>
  test
</Text>
`;

exports[`renders correctly Secondary bold 1`] = `
<Text
  bold={true}
  style={
    Array [
      Object {
        "color": "#111714",
        "fontFamily": "Inter-Medium",
        "fontSize": 16,
      },
      undefined,
      Object {
        "fontFamily": "Inter-Bold",
      },
    ]
  }
>
  test
</Text>
`;

exports[`renders correctly Secondary center 1`] = `
<Text
  center={true}
  style={
    Array [
      Object {
        "color": "#111714",
        "fontFamily": "Inter-Medium",
        "fontSize": 16,
      },
      undefined,
      Object {
        "textAlign": "center",
      },
    ]
  }
>
  test
</Text>
`;

exports[`renders correctly Secondary darkGray 1`] = `
<Text
  darkGray={true}
  style={
    Array [
      Object {
        "color": "#111714",
        "fontFamily": "Inter-Medium",
        "fontSize": 16,
      },
      undefined,
      Object {
        "color": "#585D5B",
      },
    ]
  }
>
  test
</Text>
`;

exports[`renders correctly Secondary green 1`] = `
<Text
  green={true}
  style={
    Array [
      Object {
        "color": "#111714",
        "fontFamily": "Inter-Medium",
        "fontSize": 16,
      },
      undefined,
      Object {
        "color": "#4BB377",
      },
    ]
  }
>
  test
</Text>
`;

exports[`renders correctly Secondary lightGray 1`] = `
<Text
  lightGray={true}
  style={
    Array [
      Object {
        "color": "#111714",
        "fontFamily": "Inter-Medium",
        "fontSize": 16,
      },
      undefined,
      Object {
        "color": "#9CA6A0",
      },
    ]
  }
>
  test
</Text>
`;

exports[`renders correctly Secondary white 1`] = `
<Text
  style={
    Array [
      Object {
        "color": "#111714",
        "fontFamily": "Inter-Medium",
        "fontSize": 16,
      },
      undefined,
      Object {
        "color": "#FFFFFF",
      },
    ]
  }
  white={true}
>
  test
</Text>
`;

exports[`renders correctly Tertiary 1`] = `
<Text
  style={
    Array [
      Object {
        "color": "#111714",
        "fontFamily": "Inter-Medium",
        "fontSize": 14,
      },
      undefined,
    ]
  }
>
  test
</Text>
`;

exports[`renders correctly Tertiary bold 1`] = `
<Text
  bold={true}
  style={
    Array [
      Object {
        "color": "#111714",
        "fontFamily": "Inter-Medium",
        "fontSize": 14,
      },
      undefined,
      Object {
        "fontFamily": "Inter-Bold",
      },
    ]
  }
>
  test
</Text>
`;

exports[`renders correctly Tertiary center 1`] = `
<Text
  center={true}
  style={
    Array [
      Object {
        "color": "#111714",
        "fontFamily": "Inter-Medium",
        "fontSize": 14,
      },
      undefined,
      Object {
        "textAlign": "center",
      },
    ]
  }
>
  test
</Text>
`;

exports[`renders correctly Tertiary darkGray 1`] = `
<Text
  darkGray={true}
  style={
    Array [
      Object {
        "color": "#111714",
        "fontFamily": "Inter-Medium",
        "fontSize": 14,
      },
      undefined,
      Object {
        "color": "#585D5B",
      },
    ]
  }
>
  test
</Text>
`;

exports[`renders correctly Tertiary green 1`] = `
<Text
  green={true}
  style={
    Array [
      Object {
        "color": "#111714",
        "fontFamily": "Inter-Medium",
        "fontSize": 14,
      },
      undefined,
      Object {
        "color": "#4BB377",
      },
    ]
  }
>
  test
</Text>
`;

exports[`renders correctly Tertiary lightGray 1`] = `
<Text
  lightGray={true}
  style={
    Array [
      Object {
        "color": "#111714",
        "fontFamily": "Inter-Medium",
        "fontSize": 14,
      },
      undefined,
      Object {
        "color": "#9CA6A0",
      },
    ]
  }
>
  test
</Text>
`;

exports[`renders correctly Tertiary white 1`] = `
<Text
  style={
    Array [
      Object {
        "color": "#111714",
        "fontFamily": "Inter-Medium",
        "fontSize": 14,
      },
      undefined,
      Object {
        "color": "#FFFFFF",
      },
    ]
  }
  white={true}
>
  test
</Text>
`;
